src/wallet/walletdb.h

```diff
@@ -43,8 +43,16 @@ class CKeyMetadata
 {
 public:
     static const int CURRENT_VERSION=1;
+    static const int VERSION_SUPPORT_FLAGS=2;
+    static const uint8_t KEY_ORIGIN_UNSET         = 0x0000;
+    static const uint8_t KEY_ORIGIN_UNKNOWN       = 0x0001;
+    static const uint8_t KEY_ORIGIN_IMPORTED      = 0x0002;
+    static const uint8_t KEY_ORIGIN_UNENC_WALLET  = 0x0004;
+    static const uint8_t KEY_ORIGIN_ENC_WALLET    = 0x0008;
```

Comment:
But an enumeration is extensible too, by adding entries. You could say that a bitfield limits extensibility as -at most- 8 flags are possible, whereas an enumeration allows for 256 in the same space.
Unless combinations are meaningful, there is no need to use a bitfield.